kernel/src/c/mem/page.c

```diff
@@ -198,15 +201,30 @@ void setPageDirectory(PageDirectory* dir) {
 }
 
 /**
- * Align a pointer to the next 4K alignment, or return it if it is aligned
+ * Align a pointer to the previous 4K alignment, or return it if it is aligned
  * @param ptr
  * @return
  */
 static inline uint32_t alignToPage(uint32_t ptr) {
-    if ((ptr & 0x0000FFF) == 0) return ptr;
-    return (ptr & 0xFFFFF000) + 0x1000;
+    if ((ptr & WITHIN_4K_MASK) == 0) {
+        return ptr;
+    }
+    return (ptr & ALIGN_4K_MASK);
 }
 
+/**
+ * Allocate a physical region where the virtual and
+ * physical addresses of memory will match. Only
+ * whole pages can be allocated this way, and only
+ * if those pages have not been allocated yet.
+ *
+ * @param address allocate starting from the page before
+ *      this address. If any memory is allocated, this
+ *      address will be valid memory.
+ * @param length length in bytes to allocate for. Note that
+ *      only whole pages will be allocated, rounded down.
```

Comment:
It's not great that this currently rounds down. I'd like to change that in the future. If the region is free, calling mmapPhysical(address, sizeNeeded) should work, without having to think about the concept of pages.